File: solutions/php/hamming/1/Hamming.php
<?php

declare(strict_types=1);

function distance(string $strandA, string $strandB): int
{
    $distance = 0;
    $len = strlen($strandA);

    if ($len != strlen($strandB))
        throw new \InvalidArgumentException("DNA strands must be of equal length.");

    for ($i = 0; $i < $len; $i++) {
        if ($strandA[$i] != $strandB[$i]) $distance++;
    }

    return $distance;
}

File: solutions/python/hamming/1/hamming.py
def distance(strand_a: str, strand_b: str) -> int:
    """Return the Hamming distance between two DNA strands."""
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")
    return sum(first != second for first, second in zip(strand_a, strand_b))
